lib/report.js

```diff
@@ -62,8 +62,23 @@ class Report {
       try {
         const path = resolve(this.resolve, v8ScriptCov.url)
         const script = v8toIstanbul(path, this.wrapperLength)
-        script.applyCoverage(v8ScriptCov.functions)
-        map.merge(script.toIstanbul())
+
+        if (v8ScriptCov.functions.every(({ functionName, ranges }) => {
+          if (!script.source.includes(functionName)) {
```

Comment:
First check: if the file doesn't contain any sequence of characters included in coverage's `functionName`, that means the coverage is generated form the CJS-ESM wrapper.